File: apps/plant_shop_c-sharp/plant_shop_asp_dapper/Models/OrderItem.cs
namespace plant_shop_asp_dapper.Models
{
    /// <summary>
    /// Ligne de commande.
    /// </summary>
    public class OrderItem
    {
        public int Id { get; set; }
        public int OrderId { get; set; }
        public int PlantId { get; set; }
        public int Quantity { get; set; }
        public decimal Price { get; set; }

        // Propriété de navigation
        public Plant? Plant { get; set; }
    }
}

File: apps/plant_shop_java/Structure_en_micro_services/plant_shop_java_quarkus_microservices/utils/src/models/OrderItem.java
package models;

import java.math.BigDecimal;

/**
 * Modele representant un article d'une commande.
 */
public final class OrderItem {
    public int id;
    public int orderId;
    public int plantId;
    public int quantity;
    public BigDecimal price;

    /**
     * Constructeur complet.
     *
     * @param id ID de l'article
     * @param orderId ID de la commande
     * @param plantId ID de la plante
     * @param quantity Quantite commandee
     * @param price Prix unitaire
     */
    public OrderItem(int id, int orderId, int plantId, int quantity, BigDecimal price) {
        this.id = id;
        this.orderId = orderId;
        this.plantId = plantId;
        this.quantity = quantity;
        this.price = price;
    }

    /**
     * Constructeur simplifie pour la creation.
     *
     * @param orderId ID de la commande
     * @param plantId ID de la plante
     * @param quantity Quantite commandee
     * @param price Prix unitaire
     */
    public OrderItem(int orderId, int plantId, int quantity, BigDecimal price) {
        this(0, orderId, plantId, quantity, price);
    }

    // Constructeur par défaut nécessaire pour la désérialisation JSON
    public OrderItem() {}
}
